### PR TITLE
fix: Updating metrics throughout the side-car

### PR DESCRIPTION
**Pull request by davidterpay**

### CODE REVIEW

File: providers/base/api/metrics/api_query_handler.go [hunk 1]
```diff
@@ -90,6 +108,27 @@ func (m *APIMetricsImpl) AddProviderResponse(providerName string, id string, sta
 	).Add(1)
 }
 
+// AddHTTPStatusCode increments the number of responses by provider and status.
```
Comment (aljo242) [inline providers/base/api/metrics/api_query_handler.go]: nit: copypasta godoc

File: providers/types/errors.go [hunk 2]
```diff
@@ -71,7 +70,7 @@ type ErrorWithCode struct {
 
 // Error returns an error string wrapping the internalErr and the error code.
 func (ec ErrorWithCode) Error() string {
-	return fmt.Sprintf("%s: %s", ec.code.Error(), ec.internalErr.Error())
+	return ec.internalErr.Error()
```
Comment (aljo242) [inline providers/types/errors.go:73]: solving the wrapped error nightmare

**Review by aljo242 — APPROVED** [review]
lgtm with nit

File: providers/types/errors.go (hunk 2 above)
Comment (davidterpay) [reply providers/types/errors.go:73]: soon

**Pull request merged**